packages/app-elements/src/ui/forms/RuleEngine/Condition/ValueComponents/InputResourceSelector.tsx

```diff
@@ -0,0 +1,136 @@
+import type { QueryParamsList, Tag } from "@commercelayer/sdk"
+import { uniqBy } from "lodash-es"
+import type React from "react"
+import { useCoreApi, useCoreSdkProvider } from "#providers/CoreSdkProvider"
+import {
+  InputSelect,
+  type InputSelectValue,
+  isMultiValueSelected,
+  isSingleValueSelected,
+} from "#ui/forms/InputSelect"
+import { useRuleEngine } from "../../RuleEngineContext"
+import type { ItemWithValue } from "../../utils"
+import type { useResourcePathInfos } from "../hooks"
+
+export const InputResourceSelector: React.FC<{
+  value?: ItemWithValue["value"]
+  pathKey: string
+  infos: ReturnType<typeof useResourcePathInfos>["infos"]
+}> = ({ value, pathKey, infos }) => {
+  const { sdkClient } = useCoreSdkProvider()
+
+  const { setPath } = useRuleEngine()
+
+  const key = infos?.field?.name ?? "id"
+
+  const resource =
```

Comment:
If possible I would use a lookup table here, to avoid nesting so many conditions and to better support new resources in the future